### PR TITLE
replace remaining read_image by decode_image

### LINKED ISSUE #3501: Feedback about Datasets & DataLoaders
There is the following issue on this page: https://docs.pytorch.org/tutorials/beginner/basics/data_tutorial.html

There is a using outdated func in example for the method __getitem__. Firstly in the section "Creating a Custom Dataset for your files" in method __getitem__ it is written `image = decode_image(img_path)` , but later in example for description of method __getitem__ there is used outdated func `image = read_image(img_path)`.

### PR DESCRIPTION
**Pull request by tttiago**
Fixes #3501

## Description
Replaced the `read_image` that was still remaining by `decode_image`.

## Checklist
- [x] The issue that is being fixed is referred in the description (see above "Fixes #ISSUE_NUMBER")
- [x] Only one issue is addressed in this pull request
- [x] Labels from the issue that this PR is fixing are added to this pull request
- [x] No unnecessary issues are included into this pull request.

### CODE REVIEW

**Comment by sekyondaMeta** [comment]
@claude
